instrumentation/mongo/lib/opentelemetry/instrumentation/mongo/middlewares/subscriber.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require_relative 'command_serializer'
+
+module OpenTelemetry
+  module Instrumentation
+    module Mongo
+      module Middlewares
+        # Event handler class for Mongo Ruby driver
+        class Subscriber
+          THREAD_KEY = :__opentelemetry_mongo_spans__
```

Comment:
Broadly speaking I know that while the Datadog Implementation relies on thread local storage for it's mongo instrumentation, I'm not sure what Otel wants to support or if we have more 1st class helpers to do this. I'd like to get some other opinions here.

Reply:
Following discussion below, this should be acceptable for now, and if in the future OTEL ruby provides an API for thread storage we can refactor in a separate PR. Resolving discussion.

Reply:
Ideally, we'd use `Context` for this, but that tends to assume consistent nesting of operations. I this case, it looks like we can't really avoid the thread-local.

Reply:
@fbogsany yep originally I actually tried using Context but then abandoned it.